Client/Frontend/Browser/SafeBrowsingHelper.swift

```diff
@@ -0,0 +1,269 @@
+        let threatEntryTypes: [ThreatEntryType]
+        
+        /// The threat entries to be checked.
+        let threatEntries: [ThreatEntry]
+    }
+    
+    /// A match when checking a threat entry in the Safe Browsing threat lists.
+    struct ThreatMatch: Codable {
+        
+        /// The threat type matching this threat.
+        let threatType: ThreatType
+        
+        /// The platform type matching this threat.
+        let platformType: PlatformType
+        
+        /// The threat entry type matching this threat.
+        let threatEntryType: ThreatEntryType
+        
+        /// The threat matching this threat.
+        let threat: ThreatEntry
+        
+        /// Optional metadata associated with this threat.
+        let threatEntryMetadata: ThreatEntryMetadata?
+        
+        /// The cache lifetime for the returned match.
+        /// Clients must not cache this response for more than this duration to avoid false positives.
+        ///
+        /// A duration in seconds with up to nine fractional digits, terminated by 's'.
+        /// Example: "3.5s".
+        let cacheDuration: String
+    }
+    
+    struct Request: Codable {
+        let client: ClientInfo
+        let threatInfo: ThreatInfo
+    }
+    
+    struct Response: Codable {
+        let matches: [ThreatMatch]
+        
+        init() {
+            matches = []
+        }
+        
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            self.matches = try container.decodeIfPresent([ThreatMatch].self, forKey: .matches) ?? []
+        }
+    }
+    
+    struct ResponseError: Codable {
+        let code: Int
+        let message: String
+        
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: NestedKeys.self)
+            let errorContainer = try container.nestedContainer(keyedBy: CodingKeys.self, forKey: .error)
+            
+            self.code = try errorContainer.decode(Int.self, forKey: .code)
+            self.message = try errorContainer.decode(String.self, forKey: .message)
+        }
+        
+        private enum NestedKeys: String, CodingKey {
+            case error
+        }
+    }
+    
+    private class SafeBrowsingError: NSError {
+        public init(_ message: String, code: Int = -1) {
+            super.init(domain: "SafeBrowsingError", code: code, userInfo: [
+                NSLocalizedDescriptionKey: message,
+                NSLocalizedFailureErrorKey: message
+                ])
+        }
+        
+        required init?(coder aDecoder: NSCoder) {
+            fatalError()
+        }
+    }
+    
+    @discardableResult
+    public static func threatMatches(urls: [URL], session: URLSession = .shared, _ completion: @escaping (Response, Error?) -> Void) throws -> URLSessionDataTask {
+        
+        let apiKey = "AIzaSyDeglae_dQQyQuRNk1jPq5R5--jBy21H5o"
```

Comment:
As a rule of thumb, never include API keys in the source code. We need to use the safe-browsing proxy for this. See: https://github.com/brave/handbook/blob/master/development/aws.md - This applies for all keys.